test/tests/functional/pbs_soft_walltime.py

```diff
@@ -275,20 +267,20 @@ def test_soft_walltime_extend(self):
         J = Job(TEST_USER)
         jid = self.server.submit(J)
 
-        self.server.alterjob(jid, {'Resource_List.soft_walltime': 5})
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 6})
```

Comment:
I think this will fix the issue you were seeing with this test.  The job was missing the 15s extension due to the expect calls.  Each expect call added 2s.  6s+6s+4s = 16s.  This caused us to miss the 15s extension and jump straight to 20s.  With 6s, we should have more leeway for the test to run.  If you still see this test fail, we can extend it further.